torch/distributed/_sharded_tensor/__init__.py

```diff
@@ -510,3 +511,109 @@ def shard_parameter(
 
     # Now we can set the attribute appropriately.
     setattr(module, param_name, st)
+
+
+def init_sharded_tensor_from_local_result(
+    sharded_tensor,
+    local_result,
+    tensor_shard_dim,
+    result_shard_dim,
+    world_size,
+    pg,
+):
+    """
+    Given a sharded tensor and local_result from an op on top of it. We want
+    to create a new sharded tensor from the local_result so that the the next
+    op can be performed on the basis of the new sharded tensor. This can seen
+    as the last step of the first phase of the Megatron-LM style model(tensor)
+    parallelism.
```

Comment:
This is a generic sharded tensor function, we shouldn't mention megatron-lm here.

Reply:
sure.